packages/plugin-essentials/sources/commands/install.ts

```diff
@@ -218,7 +218,11 @@ export default class YarnCommand extends BaseCommand {
       }
     }
 
-    const immutable = this.immutable ?? configuration.get(`enableImmutableInstalls`);
+    const updateMode = this.mode === InstallMode.UpdateLockfile;
+    if (updateMode && (this.immutable || this.immutableCache))
+      throw new UsageError(`\`--immutable\` and \`--immutable-cache\` cannot be used with \`--mode=update-lockfile\``);
```

Comment:
Instead of wrapping in backticks it would be more useful to use colors: `formatUtils.pretty(configuration, value, formatUtils.Type.CODE)`.